packages/bloc/lib/src/bloc.dart

```diff
@@ -187,3 +190,39 @@ abstract class Bloc<Event, State> extends Stream<State> implements Sink<Event> {
     onError(error, stacktrace);
   }
 }
+
+class _StartWithStreamTransformer<T> extends StreamTransformerBase<T, T> {
```

Comment:
Can we just use an async generator here as Brian mentioned? It would greatly simplify this.

Reply:
Yes, we can. I will do. What we should doing with `Bloc.isBroadcast`. After changing `_stateController` to broadcast it always return `true`? But actually bloc still support many listeners like a broadcast stream.

Reply:
We can return `true` always but it will be hardcoded value. Hmm. What do you think?

Reply:
yeah we can just hardcode true since we can just enforce that within the bloc.